IETF-RFC.md

```diff
@@ -1002,6 +1000,83 @@ Additionally, if `protocol.<protocolname>.requirements` includes
 Party has been authenticated with MFA, or prompt the consumer in order
 to elevate their session, if applicable.
 
+
+# Code Flow
+
+This section defines the procedure for issuing short-lived bearer access
+tokens for use by the Receiving Server when accessing a resource shared
+through OCM.  The mechanism is aligned with the OAuth 2.0
+*authorization_code* grant type but is performed entirely as a
+server to server interaction between the Sending and Receiving Servers.
+No user interaction or redirect is involved. [RFC6749]
+
+##  Token Request
+
+To obtain an access token, the Receiving Server MUST send an HTTP POST
+request to the Sending Server’s tokenEndPoint as discovered in the
+OCM provider metadata.
+
+```
+POST {tokenEndPoint} HTTP/1.1
+Host: my-cloud-storage.org
+Date: Wed, 05 Nov 2025 14:00:00 GMT
+Content-Type: application/x-www-form-urlencoded
+Digest: SHA-256=ok6mQ3WZzKc8nb7s/Jt2yY1uK7d2n8Zq7dhl3Q0s1xk=
+Content-Length: 101
+Signature-Input: 
+    sig1=("@method" "@target-uri" "content-digest" "date"); \
+    created=1730815200; keyid="receiver.example.org#2025"; \
+    alg="rsa-sha256"
+Signature: sig1=:
+    bM2sV2a4oM8pWc4Q8r9Zb8bQ7a2vH1kR9xT0yJ3uE4wO5lV6bZ1cP2rN3qD4tR5hC=:
+
+grant_type=authorization_code&
+client_id=receiver.example.org&
+code=ABCD1234
+```
+
+The request MUST be signed using an HTTP Message Signature
+[RFC9421].  The `client_id` identifies the Receiving Server and MUST be
+set to it’s fully qualified domain name.  The `code` parameter carries
+the authorization code that was issued by the Sending Server in the
+Share Creation Notification. It is allowed to send the additional
+parameters defined in [RFC6749] for the authorization_code grant type,
+but they MUST be ignored.
+
+##  Token Response
+
+If the request is valid and the code is accepted, the Sending Server
+MUST respond with HTTP 200 OK and a JSON object containing the issued
+token:
+
+```
+{
+  "access_token": "8f3d3f26-f1e6-4b47-9e3e-9af6c0d4ad8b",
+  "token_type": "Bearer",
```

Comment:
Isn't that implied? What would be the options?

Reply:
are you referring to the token_type? The idea in the back of my mind is to have enough OAUTH in there, to be able to reuse libraries at least focused on OAUTH, that is why I have tried to retain as much as possible the structure from OAUTH.